src/subplots.ts

```diff
@@ -1585,15 +1584,33 @@ export class BasePlot extends PlotData {
     return boolSelection
   }
 
-  public drawAxes() {
-    this.axes.forEach(axis => axis.draw(this.context_show));
+  public drawAxes() { this.axes.forEach(axis => axis.draw(this.context_show)) };
+
+  public drawSelectionRectangle(context: CanvasRenderingContext2D) {
+    // TODO: change with newRect
+    Shape.rect(this.X, this.Y, this.width, this.height, context, "hsl(203, 90%, 88%)", "hsl(0, 0%, 0%)", 1, 0.3, [15,15]);
   }
 
+  public drawMovingObjects() {}
+
+  public computeMovingObjects() {}
+
   public draw(): void {
+    this.context_show.save();
     this.drawCanvas();
     this.context_show.setTransform(this.canvasMatrix);
     this.updateAxes();
+    this.computeMovingObjects();
+
+    this.context_show.setTransform(this.movingMatrix);
+    this.drawMovingObjects();
+
+    this.context_show.setTransform(this.canvasMatrix);
     this.drawAxes();
+    
+    this.context_show.resetTransform();
+    // if (this.buttons_ON) { this.drawButtons(context) }
```

Comment:
Forgotten comment ?

Reply:
I tried to put buttons but stopped when i saw how it was done in current versions.
I let it here to remind to implement it and also where to write instructions. I can remove if required